### DISCUSSION TITLE
Can someone explain what `preview-protocol.js` does? Thank you very much!

### DISCUSSION THREAD

**Discussion by lilong7676** [codesandbox/codesandbox-client#8399] [Q&A]
I want to privatise my deployment to my own server (with no access to the external network), but I see preview-protocol.js ([link](https://github.com/codesandbox/codesandbox-client/blob/d84588e605c29c9c587fc6720baab40e613f6418/packages/sandbox-hooks/console/index.js#L63))  loaded in my browser's network request every time, what does it do? I don't understand this part of the code because it doesn't seem to be open source.

**Comment by JamesACS** [comment]
Hello @lilong7676 ,

I'm afraid we don't offer native deployment through CodeSandbox. If you're looking to deploy privately we'd recommend downloading the environment and then deploying locally.
